examples/js/main-deferred.js

```diff
@@ -0,0 +1,174 @@
+/**
+ * @file HTML Highlighter deferred example.
+ * @copyright 2015 Diffeo
+ *
+ * Comments:
+ *
+ */
+
+
+(function (factory, window) {
+
+  window.require( [ 'jquery', 'src/html_highlighter' ], function ($, hh) {
+    return factory(window, $, hh);
+  } );
+
+}(function (window, $, hh, undefined) {
+
+  var MAX_HIGHLIGHT = 5;
+
+  var dataBaseUrl = 'data/',
+      dataSources = [ {
+        name: 'Viber hacked',
+        url: 'viber-attacked-by-syrian-electronic-army.json'
+      }, {
+        name: 'Viber hacked -- cropped',
+        url: 'viber-attacked-by-syrian-electronic-army-cropped.json'
+      }, {
+        name: 'Ars Technica',
+        url: 'ars-technica.json'
+      }, {
+        name: 'Simple',
+        url: 'simple.json'
+      }, {
+        name: 'One paragraph',
+        url: 'one-para.json'
+      } ];
+
+  var elSelector = $('#filter-data'),
+      elDocument = $('#document'),
+      elWidgetSelection = $('#widget-selection'),
+      elWidgetMain = $('#widget-main'),
+      elSearch = $('#search'),
+      elAdd = $('#add');
+
+  var count = 0,
+      mouseDown = 0,
+      highlighter;
+
+
+  function init()
+  {
+    /* Set-up sequence
+     * --
+     * UI */
+    elSearch.keyup(function () {
+      elAdd.attr('disabled', elSearch.val().trim().length === 0);
+    } );
+    elSearch.val('').trigger('keyup').focus();
+
+    elAdd.click(function () {
+      var name = elSearch.val(),
+          cmd = new hh.DeferredExecution(highlighter)
```

Comment:
@miguel-guedes I like the idea you have here with `DeferredExecution` (and `Command{Add,Enable}Query`), but I think it makes the API too complex. Can "deferred execution" just be the default when you call `HtmlHighlighter.add`? Queries can be enabled by default. Instead of the user building up an explicit command and then calling `run`, they can just call `HtmlHighlighter.apply`, and that changes the UI.

Notably, if the user wants to retain the current behavior of `HtmlHighlighter.add`, they can easily implement themselves.